cohorts/load.py

```diff
@@ -397,14 +412,16 @@ def is_lambda(func):
             for key, value in on.iteritems():
                 col, df = apply_func(on=value, col=key, df=df)
                 cols.append(col)
-            return (cols, df)
         if type(on) == list:
             cols = []
             for i, elem in enumerate(on):
                 col = elem.__name__ if not is_lambda(elem) else "column_%d" % i
                 col, df = apply_func(on=elem, col=col, df=df)
                 cols.append(col)
-            return (cols, df)
+
+        if (rename_cols):
+            df = df.rename(columns=_strip_column_names(df.columns, keep_paren_contents=keep_paren_contents))
```

Comment:
I can't think of a scenario where the rename would fail on a column name in the `DataFrame` that it would not fail on in the `cols`? I agree that this feels a little hacky, and if you want to do it more cleanly feel free; fine with me either way.